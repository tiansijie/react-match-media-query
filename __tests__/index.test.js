import React from "react";
import renderer from "react-test-renderer";
import MatchMedia from "../src/index.js";

const mock = matches => () => ({
	addListener: () => {},
	removeListener: () => {},
	matches
});
it("renders correctly with react children", () => {
	window.matchMedia = mock(true);
	const tree = renderer
		.create(
			<MatchMedia query={"(min-width: 900px)"}>
				<div>iamhere</div>
			</MatchMedia>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it("renders correctly with function", () => {
	window.matchMedia = mock(true);
	const tree = renderer.create(
		<MatchMedia query={"(min-width: 900px)"}>
			{({ matches }) => {
				if (matches) {
					return <div>nowhere</div>;
				}
				return null;
			}}
		</MatchMedia>
	);
	expect(tree).toMatchSnapshot();
});

it("not render when media query not match", () => {
	window.matchMedia = mock(false);
	const tree = renderer
		.create(
			<MatchMedia query={"(min-width: 900px)"}>
				<div>iamhere</div>
			</MatchMedia>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it("not render when media query not match(function)", () => {
	window.matchMedia = mock(false);
	const tree = renderer
		.create(
			<MatchMedia query={"(min-width: 900px)"}>
				{({ matches }) => {
					if (matches) {
						return <div>nowhere</div>;
					}
					return null;
				}}
			</MatchMedia>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
