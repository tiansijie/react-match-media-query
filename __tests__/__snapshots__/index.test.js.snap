// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not render when media query not match 1`] = `null`;

exports[`not render when media query not match(function) 1`] = `null`;

exports[`renders correctly with function 1`] = `
<div>
  nowhere
</div>
`;

exports[`renders correctly with react children 1`] = `
<div>
  iamhere
</div>
`;
